graphql/src/Mu/GraphQL/Query/Parse.hs

```diff
@@ -659,52 +660,59 @@ class ValueParser (sch :: Schema') (v :: FieldType Symbol) where
 instance ValueParser sch ('TPrimitive Scientific) where
   valueParser _ _ (GQL.Float b) = pure $ FPrimitive $ fromFloatDigits b
-  valueParser _ fname _
-    = throwError $ "field '" <> fname <> "' was not of right type"
+  valueParser _ fname _         = throwError $ "field '" <> fname <> "' was not of right type"
 instance ValueParser sch ('TPrimitive Double) where
   valueParser _ _ (GQL.Float b) = pure $ FPrimitive b
-  valueParser _ fname _
-    = throwError $ "field '" <> fname <> "' was not of right type"
+  valueParser _ fname _         = throwError $ "field '" <> fname <> "' was not of right type"
 instance ValueParser sch ('TPrimitive T.Text) where
-  valueParser _ _ (GQL.String b)
-    = pure $ FPrimitive b
-  valueParser _ fname _
-    = throwError $ "field '" <> fname <> "' was not of right type"
+  valueParser _ _ (GQL.String b) = pure $ FPrimitive b
+  valueParser _ fname _          = throwError $ "field '" <> fname <> "' was not of right type"
 instance ValueParser sch ('TPrimitive String) where
-  valueParser _ _ (GQL.String b)
-    = pure $ FPrimitive $ T.unpack b
-  valueParser _ fname _
-    = throwError $ "field '" <> fname <> "' was not of right type"
+  valueParser _ _ (GQL.String b) = pure $ FPrimitive $ T.unpack b
+  valueParser _ fname _          = throwError $ "field '" <> fname <> "' was not of right type"
 instance (ValueParser sch r) => ValueParser sch ('TList r) where
-  valueParser vmap fname (GQL.List xs)
-    = FList <$> traverse (valueParser' vmap fname) xs
-  valueParser _ fname _
-    = throwError $ "field '" <> fname <> "' was not of right type"
+  valueParser vmap fname (GQL.List xs) = FList <$> traverse (valueParser' vmap fname) xs
+  valueParser _ fname _                = throwError $ "field '" <> fname <> "' was not of right type"
 instance (ValueParser sch r) => ValueParser sch ('TOption r) where
-  valueParser _ _ GQL.Null
-    = pure $ FOption Nothing
-  valueParser vmap fname v
-    = FOption . Just <$> valueParser' vmap fname v
+  valueParser _ _ GQL.Null = pure $ FOption Nothing
+  valueParser vmap fname v = FOption . Just <$> valueParser' vmap fname v
 instance (ObjectOrEnumParser sch (sch :/: sty), KnownName sty)
          => ValueParser sch ('TSchematic sty) where
-  valueParser vmap _ v
-    = FSchematic <$> parseObjectOrEnum' vmap (T.pack $ nameVal (Proxy @sty)) v
+  valueParser vmap _ v = FSchematic <$> parseObjectOrEnum' vmap (T.pack $ nameVal (Proxy @sty)) v
+instance ValueParser sch ('TPrimitive A.Value) where
+  valueParser vmap _ x = FPrimitive <$> toAesonValue vmap x
+instance ValueParser sch ('TPrimitive A.Object) where
+  valueParser vm _ (GQL.Object xs) = FPrimitive . HM.fromList <$> traverse (toPairs vm) xs
+  valueParser _ fname _            = throwError $ "field '" <> fname <> "' was not of right type"
+
+toPairs :: MonadError T.Text f => VariableMap -> GQL.ObjectField GQL.Value -> f (T.Text, A.Value)
```

Comment:
I would suggest having a better name for this: if you take the file as a whole it's not clear that this refers specifically to creating pairs for a JSON object.